File: DecisionTable_FOPtests.java
package FOPtests;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.junit.Assert;
import org.junit.Test;


public class DecisionTable_FOPtests {

	@Test
    public void test1_end_to_end_successful_ent_client() throws Exception {
		WebDriver driver = new FirefoxDriver();
	    // Navigate to a web page
	    driver.get("http://www.fop.com");
	    // Perform actions on HTML elements, entering text and submitting the form
	    WebElement customernameElement     = driver.findElement(By.name("customername"));
	    WebElement formElement        = driver.findElement(By.id("loginForm"));
	    customernameElement.sendKeys("Tim Barksdale");
	    
	    formElement.submit();        // submit by form element
	    WebDriverWait wait = new WebDriverWait(driver, 10);
	    WebElement messageElement = wait.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("loginResponse"))
	    );
	    String message                 = messageElement.getText();
	    String successMsg             = "Welcome to the fop website. You logged in successfully.";
	    Assert.assertEquals (message, successMsg);
	    
	    //Now assume the webpage starts asking questions to determine if FOP should be granted or not.
	    //we will assume the questions asked will follow the same order as the user story states
	    //First question is if an enterprise client or a platinum personal client
	    
	    WebElement entclientElement     = driver.findElement(By.name("EntClientButton"));
	    entclientElement.click();
	    
	    //Next it asks user if they have a checking account or not
	    WebElement checkingacctyesElement     = driver.findElement(By.name("CheckingYESButton"));
	    checkingacctyesElement.click();
	    
	    //Next question is if the cutomer has $200,000 or more in deposits
	    WebElement depositsElement     = driver.findElement(By.name("DepositsYESButton"));
	    depositsElement.click();
	    
	    //Next question is - does the customer already have overdraft protection?
	    WebElement alreadyfopElement     = driver.findElement(By.name("fopNoButton"));
	    alreadyfopElement.click();
	    
	    //and final question is - does the customer have fewer than five overdrafts 
	    //in the last twelve months?
	    WebElement fewerthanfiveElement     = driver.findElement(By.name("overdraftfewerYESButton"));
	    fewerthanfiveElement.click();
	    
	    //submit all of the answers
	    WebElement submitElement        = driver.findElement(By.id("submitButton"));
	    submitElement.click();
	    
	    //so now since all questions were answered correctly to approve overdraft protection,
	    //verify success message
	    WebDriverWait wait2 = new WebDriverWait(driver, 5);
	    WebElement responseElement = wait2.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("Response"))
	    );
	    String successmessage                 = responseElement.getText();
	    String expectedsuccessMsg             = "You have been approved for Free Overdraft Protection.";
	    Assert.assertEquals (successmessage, expectedsuccessMsg);
	    
	    driver.quit();
   }
	@Test
    public void test2_end_to_end_successful_platinum_client() throws Exception {
		WebDriver driver = new FirefoxDriver();
	    // Navigate to a web page
	    driver.get("http://www.fop.com");
	    // Perform actions on HTML elements, entering text and submitting the form
	    WebElement customernameElement     = driver.findElement(By.name("customername"));
	    WebElement formElement        = driver.findElement(By.id("loginForm"));
	    customernameElement.sendKeys("Tim Barksdale");
	    
	    formElement.submit();        // submit by form element
	    WebDriverWait wait = new WebDriverWait(driver, 10);
	    WebElement messageElement = wait.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("loginResponse"))
	    );
	    String message                 = messageElement.getText();
	    String successMsg             = "Welcome to the fop website. You logged in successfully.";
	    Assert.assertEquals (message, successMsg);
	    
	    //Now assume the webpage starts asking questions to determine if FOP should be granted or not.
	    //we will assume the questions asked will follow the same order as the user story states
	    //First question is if an enterprise client or a platinum personal client
	    
	    WebElement platclientElement     = driver.findElement(By.name("PlatClientButton"));
	    platclientElement.click();
	    
	    //Next it asks user if they have a checking account or not
	    WebElement checkingacctyesElement     = driver.findElement(By.name("CheckingYESButton"));
	    checkingacctyesElement.click();
	    
	    //Next question is if the cutomer has $200,000 or more in deposits
	    WebElement depositsElement     = driver.findElement(By.name("DepositsYESButton"));
	    depositsElement.click();
	    
	    //Next question is - does the customer already have overdraft protection?
	    WebElement alreadyfopElement     = driver.findElement(By.name("fopNoButton"));
	    alreadyfopElement.click();
	    
	    //and final question is - does the customer have fewer than five overdrafts 
	    //in the last twelve months?
	    WebElement fewerthanfiveElement     = driver.findElement(By.name("overdraftfewerYESButton"));
	    fewerthanfiveElement.click();
	    
	    //submit all of the answers
	    WebElement submitElement        = driver.findElement(By.id("submitButton"));
	    submitElement.click();
	    
	    //so now since all questions were answered correctly to approve overdraft protection,
	    //verify success message
	    WebDriverWait wait2 = new WebDriverWait(driver, 5);
	    WebElement responseElement = wait2.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("Response"))
	    );
	    String successmessage                 = responseElement.getText();
	    String expectedsuccessMsg             = "You have been approved for Free Overdraft Protection.";
	    Assert.assertEquals (successmessage, expectedsuccessMsg);
	    
	    driver.quit();
   }
	@Test
    public void test3_already_have_fop() throws Exception {
		WebDriver driver = new FirefoxDriver();
	    // Navigate to a web page
	    driver.get("http://www.fop.com");
	    // Perform actions on HTML elements, entering text and submitting the form
	    WebElement customernameElement     = driver.findElement(By.name("customername"));
	    WebElement formElement        = driver.findElement(By.id("loginForm"));
	    customernameElement.sendKeys("Tim Barksdale");
	    
	    formElement.submit();        // submit by form element
	    WebDriverWait wait = new WebDriverWait(driver, 10);
	    WebElement messageElement = wait.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("loginResponse"))
	    );
	    String message                 = messageElement.getText();
	    String successMsg             = "Welcome to the fop website. You logged in successfully.";
	    Assert.assertEquals (message, successMsg);
	    
	    //Now assume the webpage starts asking questions to determine if FOP should be granted or not.
	    //we will assume the questions asked will follow the same order as the user story states
	    //First question is if an enterprise client or a platinum personal client
	    
	    WebElement platclientElement     = driver.findElement(By.name("PlatClientButton"));
	    platclientElement.click();
	    
	    //Next it asks user if they have a checking account or not
	    WebElement checkingacctyesElement     = driver.findElement(By.name("CheckingYESButton"));
	    checkingacctyesElement.click();
	    
	    //Next question is if the customer has $200,000 or more in deposits
	    WebElement depositsElement     = driver.findElement(By.name("DepositsYESButton"));
	    depositsElement.click();
	    
	    //Next question is - does the customer already have overdraft protection? we answer yes for this test.
	    WebElement alreadyfopElement     = driver.findElement(By.name("fopYesButton"));
	    alreadyfopElement.click();
	    
	    //verify failure message
	    WebDriverWait wait2 = new WebDriverWait(driver, 5);
	    WebElement responseElement = wait2.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("Response"))
	    );
	    String failuremessage                 = responseElement.getText();
	    String expectedfailureMsg             = "You have been denied for Free Overdraft Protection.";
	    Assert.assertEquals (failuremessage, expectedfailureMsg);
	    
	    driver.quit();
   }
	@Test
    public void test4_no_checking() throws Exception {
		WebDriver driver = new FirefoxDriver();
	    // Navigate to a web page
	    driver.get("http://www.fop.com");
	    // Perform actions on HTML elements, entering text and submitting the form
	    WebElement customernameElement     = driver.findElement(By.name("customername"));
	    WebElement formElement        = driver.findElement(By.id("loginForm"));
	    customernameElement.sendKeys("Tim Barksdale");
	    
	    formElement.submit();        // submit by form element
	    WebDriverWait wait = new WebDriverWait(driver, 10);
	    WebElement messageElement = wait.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("loginResponse"))
	    );
	    String message                 = messageElement.getText();
	    String successMsg             = "Welcome to the fop website. You logged in successfully.";
	    Assert.assertEquals (message, successMsg);
	    
	    //Now assume the webpage starts asking questions to determine if FOP should be granted or not.
	    //we will assume the questions asked will follow the same order as the user story states
	    //First question is if an enterprise client or a platinum personal client
	    
	    WebElement platclientElement     = driver.findElement(By.name("PlatClientButton"));
	    platclientElement.click();
	    
	    //Next it asks user if they have a checking account or not. we enter a no for this test.
	    WebElement checkingacctnoElement     = driver.findElement(By.name("CheckingNOButton"));
	    checkingacctnoElement.click();
	    
	  //verify failure message
	    WebDriverWait wait2 = new WebDriverWait(driver, 5);
	    WebElement responseElement = wait2.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("Response"))
	    );
	    String failuremessage                 = responseElement.getText();
	    String expectedfailureMsg             = "You have been denied for Free Overdraft Protection.";
	    Assert.assertEquals (failuremessage, expectedfailureMsg);
	    
	    driver.quit();
   }
	@Test
    public void test5_not_enough_deposits() throws Exception {
		WebDriver driver = new FirefoxDriver();
	    // Navigate to a web page
	    driver.get("http://www.fop.com");
	    // Perform actions on HTML elements, entering text and submitting the form
	    WebElement customernameElement     = driver.findElement(By.name("customername"));
	    WebElement formElement        = driver.findElement(By.id("loginForm"));
	    customernameElement.sendKeys("Tim Barksdale");
	    
	    formElement.submit();        // submit by form element
	    WebDriverWait wait = new WebDriverWait(driver, 10);
	    WebElement messageElement = wait.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("loginResponse"))
	    );
	    String message                 = messageElement.getText();
	    String successMsg             = "Welcome to the fop website. You logged in successfully.";
	    Assert.assertEquals (message, successMsg);
	    
	    //Now assume the webpage starts asking questions to determine if FOP should be granted or not.
	    //we will assume the questions asked will follow the same order as the user story states
	    //First question is if an enterprise client or a platinum personal client
	    
	    WebElement platclientElement     = driver.findElement(By.name("PlatClientButton"));
	    platclientElement.click();
	    
	    //Next it asks user if they have a checking account or not
	    WebElement checkingacctyesElement     = driver.findElement(By.name("CheckingYESButton"));
	    checkingacctyesElement.click();
	    
	    //Next question is if the customer has $200,000 or more in deposits
	    WebElement depositsnoElement     = driver.findElement(By.name("DepositsNoButton"));
	    depositsnoElement.click();
	    
	  //verify failure message
	    WebDriverWait wait2 = new WebDriverWait(driver, 5);
	    WebElement responseElement = wait2.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("Response"))
	    );
	    String failuremessage                 = responseElement.getText();
	    String expectedfailureMsg             = "You have been denied for Free Overdraft Protection.";
	    Assert.assertEquals (failuremessage, expectedfailureMsg);
	    
	    driver.quit();
   }
	@Test
    public void test6_more_than_five_overdrafts() throws Exception {
		WebDriver driver = new FirefoxDriver();
	    // Navigate to a web page
	    driver.get("http://www.fop.com");
	    // Perform actions on HTML elements, entering text and submitting the form
	    WebElement customernameElement     = driver.findElement(By.name("customername"));
	    WebElement formElement        = driver.findElement(By.id("loginForm"));
	    customernameElement.sendKeys("Tim Barksdale");
	    
	    formElement.submit();        // submit by form element
	    WebDriverWait wait = new WebDriverWait(driver, 10);
	    WebElement messageElement = wait.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("loginResponse"))
	    );
	    String message                 = messageElement.getText();
	    String successMsg             = "Welcome to the fop website. You logged in successfully.";
	    Assert.assertEquals (message, successMsg);
	    
	    //Now assume the webpage starts asking questions to determine if FOP should be granted or not.
	    //we will assume the questions asked will follow the same order as the user story states
	    //First question is if an enterprise client or a platinum personal client
	    
	    WebElement platclientElement     = driver.findElement(By.name("PlatClientButton"));
	    platclientElement.click();
	    
	    //Next it asks user if they have a checking account or not
	    WebElement checkingacctyesElement     = driver.findElement(By.name("CheckingYESButton"));
	    checkingacctyesElement.click();
	    
	    //Next question is if the customer has $200,000 or more in deposits
	    WebElement depositsElement     = driver.findElement(By.name("DepositsYESButton"));
	    depositsElement.click();
	    
	    //Next question is - does the customer already have overdraft protection?
	    WebElement alreadyfopElement     = driver.findElement(By.name("fopNoButton"));
	    alreadyfopElement.click();
	    
	    //and final question is - does the customer have fewer than five overdrafts. for this test we say no
	    //in the last twelve months?
	    WebElement fewerthanfiveNoElement     = driver.findElement(By.name("overdraftfewerNOButton"));
	    fewerthanfiveNoElement.click();
	    
	  //verify failure message
	    WebDriverWait wait2 = new WebDriverWait(driver, 5);
	    WebElement responseElement = wait2.until(
	       ExpectedConditions.presenceOfElementLocated(By.id("Response"))
	    );
	    String failuremessage                 = responseElement.getText();
	    String expectedfailureMsg             = "You have been denied for Free Overdraft Protection.";
	    Assert.assertEquals (failuremessage, expectedfailureMsg);
	    
	    driver.quit();
	    
   }
	   
}

File: DecisionTable_FOPTests.py
import unittest
from unittest import TestCase

import pytest
import http_requests

webapp_endpoint = "www.fop.com"

username = {"username1": "Tim Barksdale"}

""" Assume this is a FOP web service that accepts posts for username strings and for answers to each question """

@pytest.fixture()
def url_http():
    return host('webapp_endpoint', scheme='http')

@pytest.fixture()
def assert_equal(is_value, expect_value):
        assert is_value == expect_value, str(is_value) + " != " + str(expect_value)

@pytest.fixture()
def expect_ok(response):
    assert response.status_code == 200, "Response status_code {0} != 200\nResponse reason: {1}\nResponse text: {2}".format(
        str(response.status_code), response.reason.encode('utf-8'), response.text.encode('utf-8'))

class FOPTests(unittest.TestCase):
    """
    All tests in this class
    """
    def test1_end_to_end_successful_ent_client(self):

        response = http_requests.get(url_http(), allow_redirects=True)
        expect_ok(response)
        assert "Welcome to FOP" in response.text

        username['username1'] = self.username[0]
        response = http_requests.post(url_http() + "username", data=username)
        assert response.json()["is_successful"], response.json()

        """ First answer if enterprise or platinum"""

        response = http_requests.post(url_http(), data=EntClientChoice)
        assert response.status_code == 200

        """ Next answer yes to the checking account question"""

        response = http_requests.post(url_http(), data=CheckingAcctYES)
        assert response.status_code == 200

        """ Next question is if the customer has $200, 000 or more in deposits """

        response = http_requests.post(url_http(), data=DepositsYES)
        assert response.status_code == 200

        """ Next question is - does the customer already have overdraft protection? """
        """ For this test answer NO"""

        response = http_requests.post(url_http(), data=OverdraftProtectionNO)
        assert response.status_code == 200

        """ and final question is - does the customer have fewer than five overdrafts in the last 12 months? """

        response = http_requests.post(url_http(), data=OverdraftFewerYES)
        assert response.status_code == 200

        """ Post a submit and verify success message"""
        response = http_requests.post(url_http(), data=submit)
        assert response.json()["You have been approved for Free Overdraft Protection."], response.json()


    def test2_end_to_end_successful_platinum_client(self):
        """
        Navigate to a web page
        """
        response = http_requests.get(url_http(), allow_redirects=True)
        expect_ok(response)
        assert "Welcome to FOP" in response.text

        username['username1'] = self.username[0]
        response = http_requests.post(url_http() + "username", headers=None, data=username)
        assert response.json()["is_successful"], response.json()

        """ First answer if enterprise or platinum"""

        response = http_requests.post(url_http(), data=PlatinumClientChoice)
        assert response.status_code == 200

        """ Next answer yes to the checking account question"""

        response = http_requests.post(url_http(), data=CheckingAcctYES)
        assert response.status_code == 200

        """ Next question is if the customer has $200, 000 or more in deposits """

        response = http_requests.post(url_http(), data=DepositsYES)
        assert response.status_code == 200

        """ Next question is - does the customer already have overdraft protection? """
        """ For this test answer NO"""

        response = http_requests.post(url_http(), data=OverdraftProtectionNO)
        assert response.status_code == 200

        """ and final question is - does the customer have fewer than five overdrafts in the last 12 months? """

        response = http_requests.post(url_http(), data=OverdraftFewerYES)
        assert response.status_code == 200

        """ Post a submit and verify success message"""
        response = http_requests.post(url_http(), data=submit)
        assert response.json()["You have been approved for Free Overdraft Protection."], response.json()



    def test3_already_have_fop(self):
        """
        Navigate to a web page
        """
        response = http_requests.get(url_http(), allow_redirects=True)
        expect_ok(response)
        assert "Welcome to FOP" in response.text

        username['username1'] = self.username[0]
        response = http_requests.post(url_http() + "username", headers=None, data=username)
        assert response.json()["is_successful"], response.json()

        """ First answer if enterprise or platinum"""

        response = http_requests.post(url_http(), data=EntClientChoice)
        assert response.status_code == 200

        """ Next answer yes to the checking account question"""

        response = http_requests.post(url_http(), data=CheckingAcctYES)
        assert response.status_code == 200

        """ Next question is if the customer has $200, 000 or more in deposits """

        response = http_requests.post(url_http(), data=DepositsYES)
        assert response.status_code == 200

        """ Next question is - does the customer already have overdraft protection? """
        """ For this test answer YES"""

        response = http_requests.post(url_http(), data=OverdraftProtectionYES)
        assert response.json()["Free Overdraft Protection Denied."], response.json()


    def test4_no_checking(self):
        """
        Navigate to a web page
        """
        response = http_requests.get(url_http(), allow_redirects=True)
        expect_ok(response)
        assert "Welcome to FOP" in response.text

        username['username1'] = self.username[0]
        response = http_requests.post(url_http() + "username", headers=None, data=username)
        assert response.json()["is_successful"], response.json()

        """ First answer if enterprise or platinum"""

        response = http_requests.post(url_http(), data=EntClientChoice)
        assert response.status_code == 200

        """ Next answer NO to the checking account question for this test"""

        response = http_requests.post(url_http(), data=CheckingAcctNO)
        assert response.json()["Free Overdraft Protection Denied."], response.json()


    def test5_not_enough_deposits(self):
        """
        Navigate to a web page
        """
        response = http_requests.get(url_http(), allow_redirects=True)
        expect_ok(response)
        assert "Welcome to FOP" in response.text

        username['username1'] = self.username[0]
        response = http_requests.post(url_http() + "username", headers=None, data=username)
        assert response.json()["is_successful"], response.json()

        """ First answer if enterprise or platinum"""

        response = http_requests.post(url_http(), data=PlatinumClientChoice)
        assert response.status_code == 200

        """ Next answer yes to the checking account question"""

        response = http_requests.post(url_http(), data=CheckingAcctYES)
        assert response.status_code == 200

        """ Next question is if the customer has $200, 000 or more in deposits, for this test answer NO """

        response = http_requests.post(url_http(), data=DepositsNO)
        assert response.json()["Free Overdraft Protection Denied."], response.json()


    def test6_more_than_five_overdrafts(self):
        """
        Navigate to a web page
        """
        response = http_requests.get(url_http(), allow_redirects=True)
        expect_ok(response)
        assert "Welcome to FOP" in response.text

        username['username1'] = self.username[0]
        response = http_requests.post(url_http() + "username", headers=None, data=username)
        assert response.json()["is_successful"], response.json()

        """ First answer if enterprise or platinum"""

        response = http_requests.post(url_http(), data=EntClientChoice)
        assert response.status_code == 200

        """ Next answer yes to the checking account question"""

        response = http_requests.post(url_http(), data=CheckingAcctYES)
        assert response.status_code == 200

        """ Next question is if the customer has $200, 000 or more in deposits """

        response = http_requests.post(url_http(), data=DepositsYES)
        assert response.status_code == 200

        """ Next question is - does the customer already have overdraft protection? """
        """ For this test answer NO"""

        response = http_requests.post(url_http(), data=OverdraftProtectionNO)
        assert response.status_code == 200

        """ and final question is - does the customer have fewer than five overdrafts in the last 12 months? """
        """ for this test answer NO"""

        response = http_requests.post(url_http(), data=OverdraftFewerNO)
        assert response.json()["Free Overdraft Protection Denied."], response.json()